contracts/Core/AssetManager.sol

```diff
@@ -34,34 +35,35 @@ contract AssetManager is ACL, AssetStorage, Constants {
         string url
     );
 
-    event JobUpdated(uint8 id, uint32 epoch, uint256 timestamp, string selector, string url);
+    event JobUpdated(uint8 id, uint32 epoch, int8 power, uint256 timestamp, string selector, string url);
 
     event JobActivityStatus(bool active, uint8 id, uint32 epoch, uint256 timestamp);
 
     event CollectionCreated(
         bool active,
-        assetTypes assetType,
         uint8 id,
-        uint8[] jobIDs,
-        address creator,
+        int8 power,
         uint32 epoch,
         uint32 aggregationMethod,
+        uint8[] jobIDs,
```

Comment:
I had sorted the datatype according to increasing order of size. This is a good thumb rule to facilitate tight packing. Please rearrange in increasing order of size.

Reply:
Yep, we usually put the dynamic array at the end to avoid fluctuations in between
    **for method gas 1 : 125491**
    **for method gas 2 : 125457**
For example 
```
// SPDX-License-Identifier: MIT
pragma solidity ^0.8.0;
contract StructPackingExample {
    
    struct collection1 {
        bool active; // 1 byte
        uint8 id;   // 1 byte
        uint8 assetType; // 1 byte
        uint8[] jobIDs; // lets say we have 3 jobs here, so 3 bytes
        address creator; // 20bytes
        uint32 aggregationMethod; // 4 bytes
        // 1 slot
        uint32 epoch; // 4 bytes
    } // 2 slot
    
    struct collection2 {
        bool active; // 1 byte
        uint8 id; // 1 byte
        uint8 assetType; // 1 byte
        uint32 epoch; // 4 
        uint32 aggregationMethod; // 4
        uint8[] jobIDs; // 3*4 = 12
        // 1 slot
        address creator; // 20
        // 2 slot
    }
    // Here as well there are 2 slots only, but the conversion cost that we have to bear in comb 1 to convert uint32 to uint256, is not here
    // that's why
    // It is usually suggested to keep a dynamic-sized array later, to avoid variations.
    
    collection1 c1;
    collection2 c2;

    function gas1(uint8[] memory ids) public {
       c1 = collection1(true, 1,1,ids, msg.sender, 1,1);
    }
    
     function gas2(uint8[] memory ids) public {
       c2 = collection2(true, 1,1,1,1,ids, msg.sender);
     }
}
```

Reply:
For Collection events, for ids size 3
gas 1 : 29944
gas 2 : 29910

```
// SPDX-License-Identifier: MIT
pragma solidity ^0.8.0;
contract StructPackingExample { bool active;
        uint8 id;
        uint8 assetType;
        uint8[] jobIDs;
        address creator;
        uint32 aggregationMethod;
        uint32 epoch;
        mapping(uint8 => bool) jobIDExist;
        string name;
    
    event CollectionCreated1(
        bool active,
        uint8 id,
        uint8[] jobIDs,
        address creator,
        uint32 epoch,
        uint32 aggregationMethod,
        uint256 timestamp,
        string name
    );
    
    
    event CollectionCreated2(
        bool active,
        uint8 id,
        uint32 epoch,
        uint32 aggregationMethod,
        uint8[] jobIDs,
        address creator,
        uint256 timestamp,
        string name
    );
    
    function gas1(uint8[] memory ids) public {
       emit CollectionCreated1(true, 1, ids, msg.sender, 1, 1, block.timestamp, "a");
    }
    
     function gas2(uint8[] memory ids) public {
       emit CollectionCreated2(true, 1, 1, 1, ids, msg.sender, block.timestamp, "a");
    }
}
```
```
// SPDX-License-Identifier: MIT
pragma solidity ^0.8.0;
contract StructPackingExample { bool active;
        uint8 id;
        uint8 assetType;
        uint8[] jobIDs;
        address creator;
        uint32 aggregationMethod;
        uint32 epoch;
        mapping(uint8 => bool) jobIDExist;
        string name;
    
    event CollectionCreated1(
        bool active,
        uint8 id,
        uint8[] jobIDs,
        address creator,
        uint32 epoch,
        uint32 aggregationMethod,
        uint256 timestamp,
        string name
    );
    
    
    event CollectionCreated2(
        bool active,
        uint8 id,
        uint32 epoch,
        uint32 aggregationMethod,
        uint8[] jobIDs,
        address creator,
        uint256 timestamp,
        string name
    );
    
    function gas1(uint8[] memory ids) public {
       emit CollectionCreated1(true, 1, ids, msg.sender, 1, 1, block.timestamp, "a");
    }
    
     function gas2(uint8[] memory ids) public {
       emit CollectionCreated2(true, 1, 1, 1, ids, msg.sender, block.timestamp, "a");
    }
}
```